lib/utils.js

```diff
@@ -66,16 +66,38 @@ function cloneContext(context, params = {}) {
 /**
  * Creates an error object and attaches the error code
  *
- * @param {number} code A non-zero error code
+ * @param {number} code
+ * @param {string} command The command to run
+ * @param {string[]} args Optional arguments to pass to the command
+ * @param {object} options Optional options to pass to child_process.spawn
  * @returns {Error} The error with code
  */
-function exitCodeError(code) {
-  const error = new Error(`Process exited with code ${code}`);
+function exitCodeError(code, command, args, options) {
+  const error = new Error(
+    `Process exited with code ${code}\n` +
+    `command: ${command}\n` +
+    `args: ${args}\n` +
+    `options: ${convertOptionsForError(options)}`
+  );
   error.code = code;
   return error;
 }
 
 /**
+ * Strips env values from the options array
+ *
+ * @param {string[]} args Optional arguments to pass to the command
+ */
+function convertOptionsForError(options) {
+  let envVars = {};
+  if (options && options.env) {
+    envVars = { envVars: Object.keys(options.env) };
+    delete options.env;
+  }
+  return JSON.stringify(Object.assign({}, options, envVars));
```

Comment:
Try `JSON.stringify(Object.assign({}, options, envVars), null, 2)` and see some magic.

```js
JSON.stringify(foo)
// "{"foo":1,"bar":[1,2,3],"baz":{"quz":[42]}}"
```

```js
JSON.stringify(foo, null, 2)
"{
  "foo": 1,
  "bar": [
    1,
    2,
    3
  ],
  "baz": {
    "quz": [
      42
    ]
  }
}"
```